### ISSUE TITLE
valgrind tests occasionally fail during shutdown

### ISSUE THREAD

**Issue by rgerhards** [rsyslog/rsyslog#2902]
This is in rsyslog core. 

**Update analysis in: https://github.com/rsyslog/rsyslog/issues/2902#issuecomment-437300808**

Most probably a new worker thread is scheduled when the worker pool was already shut down. It seems to not be able to actually start, and this will not properly be terminated. Most likely a "harmless" race. Still should be addressed.

In the meantime we should suppress this warning inside CI runs as there is no point in making CI fail for this known issue.

A typical report:
```
==26515== HEAP SUMMARY:
==26515==     in use at exit: 29,685 bytes in 85 blocks
==26515==   total heap usage: 321,712 allocs, 321,628 frees, 98,032,385 bytes allocated
==26515== 
==26515== 640 bytes in 1 blocks are possibly lost in loss record 35 of 41
==26515==    at 0x4C2B9B5: calloc (vg_replace_malloc.c:711)
==26515==    by 0x40128C4: _dl_allocate_tls (in /usr/lib64/ld-2.17.so)
==26515==    by 0x50547FB: pthread_create@@GLIBC_2.2.5 (in /usr/lib64/libpthread-2.17.so)
==26515==    by 0x44B88F: wtpStartWrkr (wtp.c:471)
==26515==    by 0x44B88F: wtpAdviseMaxWorkers (wtp.c:527)
==26515==    by 0x450397: qqueueAdviseMaxWorkers (queue.c:3066)
==26515==    by 0x450397: qqueueEnqMsg (queue.c:3076)
==26515==    by 0x410655: submitMsg2 (rsyslogd.c:1088)
==26515==    by 0x410874: logmsgInternal_doWrite (rsyslogd.c:857)
==26515==    by 0x410978: processImInternal (rsyslogd.c:1667)
==26515==    by 0x40E114: mainloop (rsyslogd.c:1893)
==26515==    by 0x40E114: main (rsyslogd.c:2087)
==26515==
```

**Comment by rgerhards** [comment]
This issue still occurs after #3113 but it seems to occur much less frequently.

**Comment by rgerhards** [comment]
This is another report from Ubuntu 18.04:
```
==28858==    at 0x4C31B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28858==    by 0x40134A6: allocate_dtv (dl-tls.c:286)
==28858==    by 0x40134A6: _dl_allocate_tls (dl-tls.c:530)
==28858==    by 0x5061227: allocate_stack (allocatestack.c:627)
==28858==    by 0x5061227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
```

Judging from what I see, it's probably related the thread stack allocation and/or thread local storage.

**Comment by rgerhards** [comment]
The PR already shows that the problem persists **and** it also shows no hint of rsyslog error. So let's try the `pthread_detach()` approach before we declare this to be induced by `pthreads lib`.

**Comment by rgerhards** [comment]
Problem re-occured again with the new PR. So I finally conclude this is a real false positive. I'll add supressions for the known cases.

**Comment by rgerhards** [comment]
I am going to solve this issue by not using detached workers and instead join them - at least if it does not turn out to be overly complicated.

**Comment by rgerhards** [comment]
This should finally be solved by #3282 - at least we did not experience it until now. So closing for now.
